metadata/src/main/java/com/redhat/lightblue/metadata/PredefinedFields.java

```diff
@@ -217,6 +213,82 @@ private static ParentNewChild ensureArraySize(EntityMetadata md, ArrayField arr)
         return ret;
     }
 
+    /**
+     * Returns <code>true</code> if the passed in field name is the object type, otherwise <code>false</code>.
+     * @param fieldName - field name to test.
+     * @return <code>true</code> if the passed in field name is the object type, otherwise <code>false</code>.
+     */
+    public static boolean isFieldObjectType(String fieldName){
+        if(fieldName == null){
+            return false;
+        }
+
+        return OBJECTTYPE_FIELD.equalsIgnoreCase(fieldName);
+    }
+
+    /**
+     * Returns <code>true</code> if the passed in field name is the count field for an array field,
+     * otherwise <code>false</code>.
+     * @param fieldName - field name to test.
+     * @param metadataFields - {@link Fields} from entity metadata.
+     * @return <code>true</code> if the passed in field name is the count field for an array field,
+     * otherwise <code>false</code>.
+     */
+    public static boolean isFieldAnArrayCount(String fieldName, Fields metadataFields){
+        if(!doesFieldNameMatchArrayCountPattern(fieldName)){
+            return false;
+        }
+
+        /*
+         * Ensure that the actual array field exists also, otherwise it might be a field
+         * that simply ends with a '#' character.
+         */
+        Field field = metadataFields.getField(createArrayFieldNameFromCountField(fieldName));
+        if((field != null) && (field instanceof ArrayField)){
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Creates and returns the array field name for the passed in array count field name.
+     * @param countField - array count field.
+     * @return the array field name for the passed in array count field name.
+     */
+    public static String createArrayFieldNameFromCountField(String countField){
+        if(countField == null){
```

Comment:
Not sure what you mean by only needing a single line, but the null check was redundant with doesFieldNameMatchArrayCountPattern.